### PR TITLE
feat(runtime): add support for python 3.10

### LINKED ISSUE #2136: Feature Request: Add support for Python 3.10
### Use case

[According to this thread](https://github.com/aws/aws-lambda-base-images/issues/31), the AWS Lambda team has already added support for Python 3.10 in the managed runtime.
We need to add this support in Powertools.

### Solution/User Experience

Add support for Python 3.10

### Alternative solutions

```markdown
- Run local tests
- Run e2e tests
- Deploy a project using AWS Lambda with Python 3.10 and verify that everything is working.
- Fix possible issues.
```

### Acknowledgment

- [X] This feature request meets [Lambda Powertools Tenets](https://awslabs.github.io/aws-lambda-powertools-python/latest/#tenets)
- [ ] Should this be considered in other Lambda Powertools languages? i.e. [Java](https://github.com/awslabs/aws-lambda-powertools-java/), [TypeScript](https://github.com/awslabs/aws-lambda-powertools-typescript/)

### PR DESCRIPTION
**Pull request by rubenfonseca**
**Issue number:** #2136

### Changes

> Please provide a summary of what's being changed

This PR makes the necessary changes to support the new [Python 3.10 runtime](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html). 

### User experience

> Please share what the user experience looks like before and after this change

After this change, our project will be tested againt Python 3.10, including the E2E tests.
With this PR we also fix a problem with the current E2E tests that were running pn Python 3.9 only.

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [x] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] I have performed a self-review of this change
* [x] Changes have been tested
* [x] Changes are documented
* [ ] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: tests/performance/test_metrics.py [hunk 1]
```diff
@@ -10,8 +10,8 @@
 from aws_lambda_powertools.metrics import metrics as metrics_global
 
 # adjusted for slower machines in CI too
-METRICS_VALIDATION_SLA: float = 0.002
-METRICS_SERIALIZATION_SLA: float = 0.002
+METRICS_VALIDATION_SLA: float = 0.003
+METRICS_SERIALIZATION_SLA: float = 0.003
```
Comment (rubenfonseca) [inline tests/performance/test_metrics.py]: Once I added python 3.10 the performance got slightly worse :(

File: .github/workflows/run-e2e-tests.yml [hunk 2]
```diff
@@ -40,7 +40,7 @@ jobs:
       - name: "Use Python"
         uses: actions/setup-python@v4
         with:
-          python-version: ${{ matrix.version }}
+          python-version: "${{ matrix.version }}"
```
Comment (leandrodamascena) [inline .github/workflows/run-e2e-tests.yml]: Should remove the double quote or doesn't matter?

File: tests/e2e/utils/infrastructure.py [hunk 3]
```diff
@@ -95,12 +95,12 @@ def create_lambda_functions(
...
         ```
 
-        Creating Lambda functions and override runtime to Python 3.7
+        Creating Lambda functions and override runtime to Python 3.10
 
         ```python
         from aws_cdk.aws_lambda import Runtime
 
-        self.create_lambda_functions(function_props={"runtime": Runtime.PYTHON_3_7)
+        self.create_lambda_functions(function_props={"runtime": Runtime.PYTHON_3_9)
```
Comment (leandrodamascena) [inline tests/e2e/utils/infrastructure.py]: Should be `Runtime.PYTHON_3_10?`

File: tests/performance/test_metrics.py (hunk 1 above)
Comment (leandrodamascena) [reply tests/performance/test_metrics.py]: Ruben, I reverted this change. Looks like it was a one-time glitch.

**Review by leandrodamascena — APPROVED** [review]
I only made a few comments that are not critical, but it's very good!
Great job Ruben!

File: .github/workflows/run-e2e-tests.yml (hunk 2 above)
Comment (rubenfonseca) [reply .github/workflows/run-e2e-tests.yml]: Done!

File: tests/e2e/utils/infrastructure.py (hunk 3 above)
Comment (rubenfonseca) [reply tests/e2e/utils/infrastructure.py]: Correct!

**Pull request merged**